config/initializers/zendesk.rb

```
@@ -0,0 +1,68 @@
+require 'yaml'
+require 'zendesk_api'
+
+def load_config_file
+  zendesk_config_path = Rails.root.join('config', 'zendesk.yml')
+  zendesk_template = ERB.new File.new(zendesk_config_path).read
+
+  begin
+    zendesk_config = YAML.load(zendesk_template.result(binding))[ENV['RAILS_ENV']]
+  rescue => e
+    Rails.logger.fatal 'Failed to parse zendesk yaml configuration. Exiting'
+    Rails.logger.fatal e
+    exit
+  end
+  zendesk_config
+end
+
+def zendesk_config_from_vcap
+  load_config_file
+  config = {}
+  begin
+    vcap = JSON.parse(zendesk_config['vcap_services'])
+    vcap['user-provided'].each do |elem|
+      if (elem.has_key? 'credentials') && (elem['credentials'].has_key? 'ZENDESK_USERNAME')
+        config["username"] = elem['credentials']['ZENDESK_USERNAME']
+        config["password"] = elem['credentials']['ZENDESK_PASSWORD']
+        config["api_key"] = elem['credentials']['ZENDESK_API_KEY']
+        config["end_point"] = elem['credentials']['ZENDESK_END_POINT']
+      end
+    end
+  rescue => e
+    Rails.logger.fatal 'Failed to extract zendesk creds from VCAP_SERVICES. Exiting'
+    Rails.logger.fatal zendesk_config['vcap_services']
+    Rails.logger.fatal e
```

would just reraise the error here instead of log and exit